src/fprime_gds/flask/static/js/datastore.js

```diff
@@ -57,8 +57,9 @@ class HistoryHelper {
         let _self = this;
         // Break our when no new items returned
         if (new_items.length === 0) { return; }
-        new_items.filter((item) => item.time).forEach((item) => {
-            item.datetime = timeToDate(item.time)
+        new_items.filter((item) => item.time).forEach((item, index) => {
+            item.datetime = timeToDate(item.time);
+            item.incremental_id = index + this.store.length;
```

Comment:
Perhaps it would be better to use a GUID or something else appended?  Theoretically, the store can be cleared and thus you could get a repeated incremental ID crossing the "clear" boundary.

Thoughts?

Reply:
In general, I do appreciate this fix!

Reply:
I don't understand what you would like to change, is it just the name of the variable or something more?

> Theoretically, the store can be cleared and thus you could get a repeated incremental ID crossing the "clear" boundary.

Yes, this solution is not perfect but it drastically reduce the risk of an error since the id is computed using timestamp+event_id+ incremental id.

Reply:
How about this?  Just add a monotonically increasing ID.

```
# Constructor
this.incrementor = 0

# In item handler:
this.incremental_id = _self.incrementor++;
```

This will work across clears of the index.  It is simple and easy to understand.  It will be unique across the events coming in to this specific screen.

A GUID is likely too heavy weight to make sense.